lib/deprecation_toolkit/warning.rb

```diff
@@ -2,25 +2,35 @@
 
 module DeprecationToolkit
   module Warning
-    def warn(str)
+    extend self
+
+    def deprecation_triggered?(str)
       if DeprecationToolkit::Configuration.warnings_treated_as_deprecation.any? { |warning| warning =~ str }
         ActiveSupport::Deprecation.warn(str)
```

Comment:
It seems strange that a predicate method is actually executing action. Also it is relying on the fact that `ActiveSupport::Deprecation.warn` returns a truthy value.

I think splint this in two methods and doing a `if/else` inside `Warning` and `Kernel` is clear.